Tracker/Coordinators/MainCoordinator.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  MainCoordinator.swift
+//  Tracker
+//
+//  Created by Filosuf on 30.01.2023.
+//
+
+import UIKit
+
+enum TabBarPage {
+    case trackers
+    case stats
+
+    var pageTitle: String {
+        switch self {
+        case .trackers:
+            return "Трекеры"
+        case .stats:
+            return "Статистика"
+        }
+    }
+
+    var image: UIImage? {
+        switch self {
+        case .trackers:
+            return UIImage(systemName: "record.circle.fill")
+        case .stats:
+            return UIImage(systemName: "hare.fill")
+        }
+    }
+}
+
+protocol MainCoordinator {
+    func startApplication() -> UIViewController
+}
+
+final class MainCoordinatorImp: MainCoordinator {
+
+    private let controllersFactory = ViewControllersFactory()
+    private var userEmail: String?
+
+    // проверка авторизован ли юзер
+    // показать либо экран авторизации, либо новостную ленту
+    func startApplication() -> UIViewController {
+        return getTabBarController()
+    }
+
+    //MARK: - Metods
+    private func getTabBarController() -> UIViewController {
+        let tabBarVC = UITabBarController()
+        tabBarVC.tabBar.backgroundColor = .white
+        let pages: [TabBarPage] = [.trackers, .stats]
+
+        tabBarVC.setViewControllers(pages.map { getNavController(page: $0) }, animated: true)
```

Comment:
красиво конечно, но в погоне за локаничностью обычно еще теряется читаемость: лучше в две строки 
```suggestion
        let viewControllers = TabBarPage.allCases.map { getNavController(page: $0) }
        tabBarVC.setViewControllers(viewControllers, animated: true)
```
TabBarPage можно подписать под протокол `CaseIterable`, чтобы было доступно свойство allCases

Reply:
Спасибо, с `allCases` интересно. Исправил